l10n_de_tax_statement/views/l10n_de_tax_statement_view.xml

```diff
@@ -0,0 +1,140 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 OpenBIG (<http://www.openbig.org>)
+     Copyright 2019 Onestein (<https://www.onestein.eu>)     
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+    <record id="view_l10n_de_tax_report_form" model="ir.ui.view">
+        <field name="model">l10n.de.tax.statement</field>
+        <field name="arch" type="xml">
+            <form>
+                <header>
+                    <field name="state" widget="statusbar" statusbar_visible="draft,posted,final"/>
+                </header>
+                <sheet>
+                    <div class="oe_button_box" name="button_box">
+                        <button name="post" string="Post" states="draft" type="object" class="oe_stat_button" icon="fa-arrow-right text-success"/>
+                        <button name="reset" string="Reset to Draft" states="posted" type="object" class="oe_stat_button" icon="fa-arrow-left text-success"/>
+                        <button name="finalize" string="Finalize" states="posted" type="object" class="oe_stat_button" icon="fa-stop-circle-o text-success" confirm="If you confirm, it will be not possible to modify this Statement or reset it back to draft anymore. Do you confirm?"/>
+                    </div>
+                    <label for="name"/>
+                    <h1>
+                        <field name="name"/>
+                    </h1>
+                    <group>
+                        <group name="tax_report">
+                            <field name="company_id" options="{'no_create': True}" groups="base.group_multi_company"/>
+                            <field name="date_range_id" attrs="{'readonly': [('state','=','posted')]}"/>
```

Comment:
You should also make the fields readonly when state is final.
See lines https://github.com/OCA/l10n-netherlands/blob/11.0/l10n_nl_tax_statement/views/l10n_nl_vat_statement_view.xml#L27-L29

Reply:
I have applied your proposed change. Thanks a lot for your intervention!